<a href="https://colab.research.google.com/github/Hagermaher/FlyRank_ML_Intership/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Refresh / Content Opportunity Scoring**
I think it's a real-world goal to know which content needs to update according to User search historical data   also Content publishing is a continuous process rather than a one-time task .SO that ML useful here that large companies can't do all that revesion manually.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Descision**: which pages should be Re-frish first .

**Who acts on it** : The decision will affect Content editors, SEO specialists, and website stakeholders who work on updating content.

**Cost**: i will answer in 2 cases :

1-False positive:if the page is good and well-appeared in searches these will cost stakeholders un-needed updates and unnecessary use of resources, also it well be time wasting and that will affect other pages that has a priority to be updated.

2-False negative: based on that the Content editors, SEO specialists, and website stakeholders won't update the content so that the content will be out-dated and it will reduce users clicks for pages ,then lower in traffic, ,so in commercial websites that can cause customer churn .

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
    subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)
elif os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("..")  # move from notebooks/ to the repo root

print("Working dir:", os.getcwd())
assert os.path.exists("data/raw/content_refresh_anonymized.csv"), "starter CSV not found — are you at the repo root?"
print("Starter data found. You're ready.")



Working dir: /content/flyrank-ml-internship-starter
Starter data found. You're ready.


In [ ]:
df=pd.read_csv("data/raw/content_refresh_anonymized.csv")
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [ ]:
df["ctr"].describe()

,ctr
count,30000.000000
mean,0.510733
std,3.279162
min,0.000000
25%,0.000000
50%,0.070000
75%,0.290000
max,100.000000


In [ ]:
df["trend_direction"].value_counts()

,count
trend_direction,
down,16262
stable,5962
up,4388
new,2236
flat,1152


In [ ]:
df["avg_position"].describe()

,avg_position
count,30000.00000
mean,16.34238
std,15.21679
min,0.00000
25%,6.20000
50%,10.80000
75%,22.30000
max,245.00000


**ANSWER**: depend on the initial statistics of the 3 column which in CTR column say the median is 0.07 which mean 50% is less than that number which suggests that some pages may deserve further review and in trend direction column most 16262 of 30000 is down  and it's the largest number which make this these support our decision to refresh content , finally in avg position median is around 10 which support that may pages need to refresh or review .These statistics suggest that there are enough signals to justify building a content opportunity scoring approahc and this lane is worth working on.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**what I can claim**:

1-This page is a good candidate for review/refresh based on multiple signals(depend on several signals not only one signal)
2-I have observed that more than 50% of pages trend_direction is down.
3-Some pages should be prioritized over others.

**What I can't claim**:

1-Refreshing page will make GOOGLE improve it's position
2-only small value in CTR make page need refresh


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.